src/features/annotation-thread/AnnotationThreadContent.js

```diff
@@ -0,0 +1,180 @@
+                    errorSubHeader: commonMessages.defaultErrorMaskSubHeaderMessage,
+                },
+            });
+
+            errorCallback(e, code, {
+                error: e,
+            });
+        },
+        [errorCallback],
+    );
+
+    const fetchAnnotation = React.useCallback(() => {
+        api.getAnnotationsAPI(false).getAnnotation(
+            fileId,
+            annotationId,
+            filePermissions,
+            getAnnotationSuccess,
+            annotationErrorCallback,
+            true,
+        );
+    }, [annotationErrorCallback, annotationId, api, fileId, filePermissions]);
+
+    React.useEffect(() => {
+        fetchAnnotation();
+    }, [annotationId, fetchAnnotation]);
+
+    const annotationSuccessCallback = () => {
+        setAnnotation({ ...annotation, isPendind: false });
+        fetchAnnotation();
+    };
+
+    const annotationDeleteSuccessCallback = () => {
+        setAnnotation({ ...annotation, isPendind: false });
+    };
+
+    const handleAnnotationDelete = ({ id, permissions }: { id: string, permissions: AnnotationPermission }) => {
+        setAnnotation({ ...annotation, isPendind: true });
+        api.getAnnotationsAPI(false).deleteAnnotation(
+            fileId,
+            id,
+            permissions,
+            annotationDeleteSuccessCallback,
+            annotationErrorCallback,
+        );
+    };
+
+    const handleAnnotationEdit = (id: string, text: string, permissions: AnnotationPermission) => {
+        setAnnotation({ ...annotation, isPendind: true });
+        api.getAnnotationsAPI(false).updateAnnotation(
+            fileId,
+            id,
+            permissions,
+            { message: text },
+            annotationSuccessCallback,
+            annotationErrorCallback,
+        );
+    };
+
+    const handleAnnotationResolve = (id: string, status: FeedItemStatus, permissions: AnnotationPermission) => {
+        setAnnotation({ ...annotation, isPendind: true });
+        api.getAnnotationsAPI(false).updateAnnotation(
+            fileId,
+            id,
+            permissions,
+            { status },
+            annotationSuccessCallback,
+            annotationErrorCallback,
+        );
+    };
+
+    /**
+     * File @mention contacts fetch success callback
+     *
+     * @private
+     * @param {BoxItemCollection} collaborators - Collaborators response data
+     * @return {void}
+     */
+    const getMentionContactsSuccessCallback = (collaborators: { entries: SelectorItems<> }): void => {
+        setMentionSelectorContacts(collaborators.entries);
+    };
+
+    const getMentions = (searchStr: string) =>
+        getCollaboratorsWithQuery(fileId, getMentionContactsSuccessCallback, errorCallback, searchStr);
+
+    const getAvatarUrl = async (userId: string): Promise<?string> => {
+        return api.getUsersAPI(false).getAvatarUrlWithAccessToken(userId, fileId);
+    };
+
+    if (annotationError) {
+        return null;
```

Comment:
This this a placeholder for a future error state?

Reply:
more like for create new annotation state. If there is no annotationId, component should render form and handle create new annotation.